### DISCUSSION TITLE
Traffic lights don't work as they should

### DISCUSSION THREAD

**Discussion by CHNtentes** [carla-simulator/carla#6947] [Q&A]
I find that at crossroads, traffic lights don't work the same as real life.

At least in my country, if 4 lights are given number like this:
```
   1
2     4
   3
```
Then 1 and 3 will turn green (roughly) at the same time, while 2 and 4 stay red. After some time, 2 and 4 turn green while 1 and 3 turn red.

However, in carla when light 1 is green, other 3 lights are all red. At any time, there is only 1 green light at crossroads. As a result, vehicles often spend a long time waiting for green light.

Is there any method to fix this behaviour?

**Comment by Jinviz** [comment]
Yes, it's possible. 
 To do that, you can first build the data of the signal flow you're actually trying to apply,
 import the traffic light information and groups from the map, 
and apply the data to the traffic light. For example, 
I built the signal flow as below.

```
<TrafficPhase Name="Phase_1" Duration="0">
		<TrafficSignalState TrafficSignalID="T1" State="Green"/>
		<TrafficSignalState TrafficSignalID="T2" State="Red"/>
		<TrafficSignalState TrafficSignalID="T3" State="Red"/>
</TrafficPhase>
<TrafficPhase Name="Phase_2" Duration="10">
		<TrafficSignalState TrafficSignalID="T1" State="Yellow"/>
		<TrafficSignalState TrafficSignalID="T2" State="Red" />
		<TrafficSignalState TrafficSignalID="T3" State="Red"/>
</TrafficPhase>
```
